File: 83.删除排序链表中的重复元素.py
#
# @lc app=leetcode.cn id=83 lang=python3
#
# [83] 删除排序链表中的重复元素
#

# @lc code=start
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return head
        index = head
        nextDiffNode = head.next
        while index:
            if not nextDiffNode:
                index.next = nextDiffNode
                index = nextDiffNode
            elif index.val != nextDiffNode.val:
                index.next = nextDiffNode
                index = nextDiffNode
                nextDiffNode = nextDiffNode.next
            else:
                nextDiffNode = nextDiffNode.next
        return head

# @lc code=end

File: 83.删除排序链表中的重复元素.java
/*
 * @lc app=leetcode.cn id=83 lang=java
 *
 * [83] 删除排序链表中的重复元素
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        if(head == null){
            return head;
        }
        ListNode index = head;
        ListNode nextDi = head.next;
        while(index!=null){
            if(nextDi==null){
                index.next = nextDi;
                index = nextDi;
            }else if(nextDi.val != index.val){
                index.next = nextDi;
                index = nextDi;
                nextDi = nextDi.next;
            }else{
                nextDi = nextDi.next;
            }
        }
        return head;
    }
}
// @lc code=end

